src/main/drivers/rcdevice_osd.c

```diff
@@ -0,0 +1,262 @@
+{
+    UNUSED(displayPort);
+    osdResetAlarms();
+    resumeRefreshAt = 0;
+    return 0;
+}
+
+int rcdeviceOSDRelease(displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return 0;
+}
+
+int rcdeviceOSDDrawScreen(displayPort_t *displayPort)
+{
+#ifdef USE_PARTICLE_DRAW
+    static uint16_t pos = 0;
+    int k = 0;
+
+    if (!rcdeviceOSDLock) {
+        rcdeviceOSDLock = true;
+
+        uint8_t data[60];
+        uint8_t dataLen = 0;
+        for (k = 0; k < MAX_CHARS2UPDATE; k++) {
+            if (screenBuffer[pos] != shadowBuffer[pos]) {
+                shadowBuffer[pos] = screenBuffer[pos];
+                uint8_t x = pos % columnCount;
+                uint8_t y = pos / columnCount;
+                data[dataLen++] = x;
+                data[dataLen++] = y;
+                data[dataLen++] = screenBuffer[pos];
+            }
+
+            if (++pos >= maxScreenSize) {
+                pos = 0;
+                break;
+            }
+        }
+        runcamDeviceDispWriteChars(osdDevice, data, dataLen);
+
+        rcdeviceOSDLock = false;
+    }
+#endif
+    return 0;
+}
+
+int rcdeviceOSDWriteString(displayPort_t *displayPort, uint8_t x, uint8_t y,
+                           const char *buff)
+{
+    UNUSED(displayPort);
+#if !defined(USE_PARTICLE_DRAW)
+    runcamDeviceDispWriteHortString(osdDevice, x, y, buff);
+#else
+    uint8_t i = 0;
+    for (i = 0; *(buff + i); i++)
+        if (x + i < columnCount) // Do not write over screen
+            screenBuffer[y * columnCount + x + i] = *(buff + i);
+#endif
+    return 0;
+}
+
+int rcdeviceOSDWriteChar(displayPort_t *displayPort, uint8_t x, uint8_t y,
+                         uint8_t c)
+{
+    UNUSED(displayPort);
+#if !defined(USE_PARTICLE_DRAW)
+    runcamDeviceDispWriteChar(osdDevice, x, y, c);
+#else
+    screenBuffer[y * columnCount + x] = c;
+#endif
+
+    return 0;
+}
+
+int rcdeviceOSDClearScreen(displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+
+#if defined(USE_PARTICLE_DRAW)
+    uint16_t x;
+    uint32_t *p = (uint32_t *)&screenBuffer[0];
+    for (x = 0; x < VIDEO_BUFFER_CHARS_PAL / 4; x++)
+        p[x] = 0x20202020;
+#else
+    runcamDeviceDispFillRegion(osdDevice, 0, 0, 255, 255, ' ');
+#endif
+
+    return 0;
+}
+
+bool rcdeviceOSDIsTransferInProgress(const displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return false;
+}
+
+int rcdeviceOSDHeartbeat(displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return 0;
+}
+
+void rcdeviceOSDResync(displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+
+    if (video_system == VIDEO_SYSTEM_PAL) {
+        displayPort->rows = RCDEVICE_PROTOCOL_OSD_VIDEO_LINES_PAL;
+    } else {
+        displayPort->rows = RCDEVICE_PROTOCOL_OSD_VIDEO_LINES_NTSC;
+    }
+
+    displayPort->cols = columnCount;
+    maxScreenSize = displayPort->rows * displayPort->cols;
+}
+
+uint32_t rcdeviceOSDTxBytesFree(const displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return UINT32_MAX;
```

Comment:
This may cause problems (converting to `int` will give -1).
Using INT_MAX or size of video buffer is safer